### PR TITLE
Ports: Add libslirp version 4.7.0

### PR DESCRIPTION
**Pull request by kennethmyhra**
The `slirp` submodule was removed from QEMU in version 7.2, running Serenity in QEMU on a Serenity host could fail without this one if running with network enabled. Although, according to timschumi, network is broken for other reasons. Even so, a `liblslirp` port will be needed when we can run Serenity on Serenity with network enabled.

For more information on the removal of the slirp submodule see the following link: https://wiki.qemu.org/ChangeLog/7.2#Removal_of_the_%22slirp%22_submodule_(affects_%22-netdev_user%22)

### CODE REVIEW

File: Ports/libslirp/package.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='libslirp'
+version='4.7.0'
+workdir="libslirp-v${version}"
+files="https://gitlab.freedesktop.org/slirp/libslirp/-/archive/v${version}/libslirp-v${version}.tar.gz libslirp-v${version}.tar.xz 9398f0ec5a581d4e1cd6856b88ae83927e458d643788c3391a39e61b75db3d3b"
+auth_type='sha256'
+useconfigure='true'
+configopts=("--cross-file" "${SERENITY_BUILD_DIR}/meson-cross-file.txt")
+depends=("glib")
```
Comment (GMTA) [inline Ports/libslirp/package.sh:9]: Non-interpolated strings should use single quotes instead of double quotes.

File: Ports/qemu/package.sh [hunk 2]
```diff
@@ -12,8 +12,14 @@ configopts=(
     "--enable-sdl"
     "--enable-tcg"
     "--enable-tools"
+    "--enable-slirp"
+)
+depends=(
+  "glib"
+  "pixman"
+  "SDL2"
+  "libslirp"
```
Comment (GMTA) [inline Ports/qemu/package.sh:21]: These should probably be in alphabetical order to prevent :redlinus: - also, single quotes.

**Comment by kennethmyhra** [comment]
Done, :redlinus: avoided

File: Ports/libslirp/package.sh (hunk 1 above)
Comment (GMTA) [reply Ports/libslirp/package.sh:9]: This doesn't look resolved?

Comment (kennethmyhra) [reply Ports/libslirp/package.sh]: Ooops, missed this one. Fixed now

File: Ports/qemu/package.sh [hunk 3]
```diff
@@ -3,17 +3,23 @@ port='qemu'
...
-    "--cross-prefix=${SERENITY_ARCH}-pc-serenity-"
-    "--extra-ldflags=-lm"
-    "--without-default-features"
-    "--disable-strip"
-    "--enable-pie"
-    "--enable-sdl"
-    "--enable-tcg"
-    "--enable-tools"
+    '--target-list=aarch64-softmmu,i386-softmmu,x86_64-softmmu'
+    '--cross-prefix=${SERENITY_ARCH}-pc-serenity-'
```
Comment (BertalanD) [inline Ports/qemu/package.sh:7]: This should be in double quotes for the interpolation, though.

Comment (kennethmyhra) [reply Ports/qemu/package.sh]: Thanks for noticing, Daniel! Things went about a bit too fast..

Additionally I reformatted the `configopts` section for `libslirp` to have each entry on a separate line.

**Comment by kennethmyhra** [comment]
OK, wait a second. I'm doing every mistake possible in this PR right now..

**Comment by kennethmyhra** [comment]
OK, now I think it should be good. The two `configopts` entries for `libslirp` has been combined into one entry.

File: Ports/qemu/package.sh [hunk 4]
```diff
@@ -12,9 +12,11 @@ configopts=(
     '--enable-sdl'
     '--enable-tcg'
     '--enable-tools'
+    '--enable-slirp'
```
Comment (linusg) [inline Ports/qemu/package.sh:15]: I hate to do this but these are in A L P H A B E T I C A L order :hyperlinus:

Comment (kennethmyhra) [reply Ports/qemu/package.sh]: Nothing escapes linus' laser eyes!

**Pull request merged**